# Schwartz basic human values and synonyms

I want to compare conservation nonprofit mission, vision, and values as well as social media content (via Tweets) to the Schwartz values theory, which suggests there are ten basic values in society: self-direction, stimulation, hedonism, achievement, power, security, tradition, conformity, benevolence, and universalism.

Jones et al. (2018) use these values in their paper, "Liberal and Conservative Values: What We Can Learn from Congressional Tweets." In order to analyze the values embedded in the tweets of Congress members, they created a list of synonyms for each of the ten values and removed any that could belong in multiple value categories. To account for language on social media, they also sorted the most common words Congress members used in tweets and validated which category they belonged in by asking independent raters to sort language into each of the ten values.

Below, my goal is to read in these synonyms, clean them, and create a dictionary that I can base analyses of mission/vision/values and tweets off of. I will also use wordnet in Python to generate additional synonyms to build up the value synonym dictionary.

In [1]:
import nltk

from nltk.corpus import stopwords,wordnet


By looking at the Jones et al. (2018) paper, I already copied and pasted their original list of synonyms and saved them in text files labelled with the values they represent. You can see that the text file has all the synonyms in it. Below, I will use the text files to create a dictionary of synonyms for each value

In [2]:
##Storing all text files in a list
values_files = ["achievement.txt",
                "benevolence.txt",
                "conformity.txt",
                "hedonism.txt",
                "power.txt",
                "security.txt",
                "self_direction.txt",
                "stimulation.txt",
                "tradition.txt",
                "universalism.txt"]

In [3]:
##Creating a dictionary to store words from text files
values_dict = dict()

In [4]:
for file in values_files: #For each text file
    with open(file, "r", encoding = "Latin-1") as infile: #Read in and tokenize the file
        key=file.replace(".txt", "")
        words=infile.read()

        tokens=nltk.word_tokenize(words)


        values_dict[key] = set(t for t in tokens if t.isalpha()) #Add the list of synonyms into the dictionary using the overarching value as the key

In [5]:
##Looking at one of the values, 'achievement'
values_dict['achievement']

{'accomplish',
 'accomplished',
 'accomplishes',
 'accomplishing',
 'accomplishment',
 'achieve',
 'achieved',
 'achievement',
 'achieves',
 'achieving',
 'advance',
 'advanced',
 'advances',
 'advancing',
 'ambition',
 'ambitions',
 'ambitious',
 'attain',
 'attained',
 'attaining',
 'attainment',
 'attains',
 'beat',
 'beating',
 'beats',
 'benefit',
 'best',
 'brainy',
 'bright',
 'brilliant',
 'capabilities',
 'capability',
 'capable',
 'champion',
 'championed',
 'champions',
 'championship',
 'competence',
 'competent',
 'complete',
 'completed',
 'completes',
 'completing',
 'completion',
 'congrats',
 'congratulations',
 'develop',
 'developed',
 'developing',
 'development',
 'develops',
 'diploma',
 'efficient',
 'excel',
 'excelled',
 'excellent',
 'excelling',
 'excels',
 'finish',
 'finished',
 'finishes',
 'finishing',
 'fruition',
 'fulfillment',
 'gain',
 'gained',
 'gaining',
 'gains',
 'goal',
 'goals',
 'graduate',
 'graduated',
 'graduates',
 'graduating',
 'graduat

I have created a dictionary of values and their synonyms that reflects the research of Jones et al. (2018). To expand on their list of values synonyms, I can use wordnet to search for additional synonyms and create an extended list. 

In [6]:
##How to make a list of synonyms for one word
syn = list()

for synset in wordnet.synsets("accomplish"):
    for lemma in synset.lemmas():
        syn.append(lemma.name())    #add the synonyms
        



In [7]:
syn

['carry_through',
 'accomplish',
 'execute',
 'carry_out',
 'action',
 'fulfill',
 'fulfil',
 'achieve',
 'accomplish',
 'attain',
 'reach']

In [8]:
#Creating a new dictionary for the extended synonym list
values_extended_dict = dict()

Below I created a for loop that takes each word under each value in the dictionary and finds synonyms for the word. The set of synonyms are added to a new extended dictionary that used the same keys as the original dictionary.

In [9]:
for value in values_dict.keys():
    syn=list() #Create an empty list for synonyms
    key=value
    for word in values_dict[value]:
        
        for synset in wordnet.synsets(word): #Find the synonums of each word in the dictionary
            
            for lemma in synset.lemmas():
                syn.append(lemma.name())    #add the synonyms to a list


            values_extended_dict[key] = syn #Add the list of synonyms to the extended dictionary using the key from the original dictionary
    
    
    

In [10]:
##The below code proves that the dictionary values under key 'conformity' are in fact different between the two dictionaries
values_extended_dict['conformity']== values_dict['conformity']

False

In [11]:
#Looking at what the for loop created by inspecting the value 'conformity'
#values_extended_dict['conformity']

In [12]:
#Are there any repeats?
print(len(values_extended_dict['conformity']))
      
print(len(set(values_extended_dict['conformity'])))

2398
474


When I print the values under 'conformity' to the screen, you can see that there is a large list of synonyms. Now, I need to clean up these synonyms by making a set of each value. While there are a ton of synonyms, a lot of them are repeats. Then, I will write each value to a new text file. 

In [13]:

for value in values_extended_dict.keys(): #For each value in the dictionary
    
    syn_unique = list(set(values_extended_dict[value])) #Create a unique set of values synonyms
        
    with open(f'{value}_synonyms.txt','w', encoding='utf-8') as out_file:  #Write them to a text file

        out_file.write("\t".join(syn_unique))


I have written 10 new text files named after each value they represent.

In [14]:
##How long is each set of synonyms?
for value in values_extended_dict.keys():
    print("The length of" + " " + str({value}) + " " + "synonyms is" + " " + str(len(set(values_extended_dict[value]))))

The length of {'achievement'} synonyms is 747
The length of {'benevolence'} synonyms is 367
The length of {'conformity'} synonyms is 474
The length of {'hedonism'} synonyms is 383
The length of {'power'} synonyms is 650
The length of {'security'} synonyms is 697
The length of {'self_direction'} synonyms is 593
The length of {'stimulation'} synonyms is 539
The length of {'tradition'} synonyms is 321
The length of {'universalism'} synonyms is 340


There is a fairly big difference between the sizes of synonym lists. "Achievement" has the most unique words at 747 words, and "tradition" only has 321 words. 

Paper that I based my values dictionary and methods on:

Jones, K. L., Noorbaloochi, S., Jost, J. T., Bonneau, R., Nagler, J., & Tucker, J. A. (2018). Liberal and conservative values: What we can learn from congressional tweets. Political Psychology, 39(2), 423-443.